cmake/functions/four_c_auto_define_module.cmake

```diff
@@ -75,23 +82,42 @@ function(four_c_auto_define_module)
     endif()
   endforeach()
 
-  # Add the headers as a file set to the interface library. This will automatically add the current directory as a search directory.
-  # Since we append sources to a target successively, we need to add multiple file sets with a unique name. Use
-  # the current directory name for this.
-  # Replace non-alphanumeric characters with underscores and make the name lowercase. Drop any leading digits and underscores.
-  string(REGEX REPLACE "[^a-zA-Z0-9]" "_" _file_set_name "${CMAKE_CURRENT_SOURCE_DIR}")
-  string(TOLOWER "${_file_set_name}" _file_set_name)
-  string(REGEX REPLACE "^[0-9_]+" "" _file_set_name "${_file_set_name}")
-
+  # We want to export/install the base directory with hierarchy. At the same time we also want to include them
+  # without hierarchies at the target as 4C does.
   target_sources(
     ${_target}_deps
     INTERFACE FILE_SET
-              ${_file_set_name}
-              TYPE
               HEADERS
               FILES
               ${_headers}
+              BASE_DIRS
+              ${PROJECT_SOURCE_DIR}/src
     )
+  include(GNUInstallDirs)
+  file(RELATIVE_PATH _relative_path ${CMAKE_SOURCE_DIR}/src ${CMAKE_CURRENT_SOURCE_DIR})
+  if(_export_target_directory)
+    target_include_directories(
+      ${_target}_deps
+      INTERFACE $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}>
+                $<INSTALL_INTERFACE:${CMAKE_INSTALL_INCLUDEDIR}/${_relative_path}>
```

Comment:
Does CMake complain when running this for the `config` target? I don't yet understand why we need a special case for it. If we configure the `config` headers to lie in `${CMAKE_INSTALL_INCLUDEDIR}/config` this should work.

Reply:
Yes, this should work. I thought that we want to put that directly under `include`.

Reply:
The reason it does not work because `include/config` does not exist, and we include it in the interface set by the command above.